### ISSUE TITLE
Rootful podman machine gets stuck using rootless connection if previously there was a rootless podman machine set as default

### ISSUE THREAD

**Issue by cbr7** [containers/podman#22577]
### Issue Description

If the user has two podman machines running at the same time on WSL, one fo them rootful and the other rootless, with the rootless one being set as default, if the user deletes the rootless machine the rootful podman machine will become default, however it's rootless connection will be used, not the rootful one.

### Steps to reproduce the issue

Steps to reproduce the issue
1. Have two podman machines running on WSL one rootful and the other rootless, rootless machine should be default.
2. Stop the rootless machine and then delete it.
3. Run command `podman system connection list` and notice that the rootless connection of the remaining rootful podman machine is now default.

### Describe the results you received

Rootful podman machine will only using the rootless connection going forward, stopping and starting the machine does not help.

### Describe the results you expected

the rootful connection of the rootful podman machine should become default.

### podman info output

```yaml
If you are unable to run podman info for any reason, please provide the podman version, operating system and its version and the architecture you are running.
```

### Podman in a container

No

### Privileged Or Rootless

Privileged

### Upstream Latest Release

Yes

### Additional environment details

Issue is only reproducible on Windows operating system.

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by cbr7** [comment]
Additional note, I was also able to reproduce this on macOS.

**Comment by jakecorrenti** [comment]
I can reproduce this on linux as well. Taking a look

**Comment by Luap99** [comment]
I think this is already being fixed in #22587

**Comment by Luap99** [comment]
> I think this is already being fixed in #22587

Sorry nevermind, wrong issue
